src/Microsoft.Azure.WebJobs.Host/Scale/IProcessMetricsProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Provider for process level metrics.
+    /// </summary>
+    internal interface IProcessMetricsProvider
+    {
+        /// <summary>
+        /// Gets the total processor time for the process.
+        /// </summary>
+        TimeSpan TotalProcessorTime { get; }
```

Comment:
Rather than have these individual metrics on this interface, I'm thinking for extensibility that I should define a GetMetrics method and return a ProcessMetrics object. That way, we can add new members in the future w/o breaking changes.